odoo/tools/mail.py

```diff
@@ -36,6 +37,27 @@
      'data-publish', 'data-id', 'data-res_id', 'data-member_id', 'data-view-id'
      ])
 
+# allow base64 image
+_image_dataurl = re.compile(r'(data:image/(.+));base64,([^"\']*)', re.I)
+_is_possibly_malicious_scheme = re.compile(r'^(?:javascript|jscript|livescript|vbscript|data|about|mocha):', re.I).search
+def _is_javascript_scheme(s):
+    if _image_dataurl.search(s):
+        return None
+    return _is_possibly_malicious_scheme(s)
+lxml.html.clean._is_javascript_scheme = _is_javascript_scheme
+
+
+def body_image_base64_to_boundary(body, callback_base64_to_name):
+    boundaries = {}
+    def base64_to_boundary(match):
+        boundary = boundaries.get(match.group(0))
+        if not boundary:
+            fileType = match.group(2)
+            name = callback_base64_to_name('boundary-%s.%s' % (len(boundaries), fileType), match.group(3))
```

Comment:
Suggested name for the images:  `'image1', 'image2', ...`. This way you don't need to extract the type and you have no security risk. The mimetype will be autodetected when the attachment is created.